### ISSUE TITLE
azurerm_cosmosdb_account ip_range_filter doesn't actually allow /8 in CIDR

### ISSUE THREAD

**Issue by stefan-kiss** [hashicorp/terraform-provider-azurerm#18134]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v1.2.8

### AzureRM Provider Version

v3.20.0

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account

### Terraform Configuration Files

```hcl
resource "azurerm_cosmosdb_account" "db" {
  name                              = "test-cosmos"
  location                          = "eastus2"
  resource_group_name               = "whatever-group-name"
  offer_type                        = "Standard"
  public_network_access_enabled     = true
  ip_range_filter                   = "0.0.0.0,1.0.0.0/8"

  enable_automatic_failover = true

  capabilities {
    name = "EnableAggregationPipeline"
  }

  capabilities {
    name = "mongoEnableDocLevelTTL"
  }


  consistency_policy {
    consistency_level       = "BoundedStaleness"
    max_interval_in_seconds = 300
    max_staleness_prefix    = 100000
  }

  geo_location {
    location          = "eastus2"
    failover_priority = 0
  }

}
```

### Debug Output/Panic Output

```shell
╷
│ Error: invalid value for ip_range_filter (Cosmos DB ip_range_filter must be a set of CIDR IP addresses separated by commas with no spaces: '10.0.0.1,10.0.0.2,10.20.0.0/16')
│
│   with module.cosmosdb.azurerm_cosmosdb_account.db,
│   on cosmosdb/main.tf line 57, in resource "azurerm_cosmosdb_account" "db":
│   57:   ip_range_filter                   = "0.0.0.0,1.0.0.0/8"
│
╵
```

### Expected Behaviour

to create the cluster with the filter requested

### Actual Behaviour

the error

notes

- azure does not accept private ip ranges there. maybe documentation should be updated 
- /8 cidr may be almost unusable - BUT it's a valid cidr and thus supported.
- the problem comes from the regexp used here : [cosmosdb_account_resource.go](https://github.com/hashicorp/terraform-provider-azurerm/blob/6109fdd95dad02bb7f83dd46a1c10e47e1314f36/internal/services/cosmos/cosmosdb_account_resource.go#L209) - the capture group for the mask has mandatory TWO digits. it could have the first digit optional (*)

### Steps to Reproduce

you need to have an existing whatever-group-name

### References

This reference is only provided as support for the statement "azure does not accept private ip ranges there"
https://github.com/hashicorp/terraform-provider-azurerm/issues/7946

**Comment by bart-vmware** [comment]
I'm being blocked by this bug as well. Trying to use: `31.0.0.0/7,106.0.0.0/6`, which is rejected inappropriately. Suggested fix is to change:

https://github.com/hashicorp/terraform-provider-azurerm/blob/00e020cbd2dd908b84cc20dbc31e9e4bb3f3f8c2/internal/services/cosmos/cosmosdb_account_resource.go#L272

to:
```go
regexp.MustCompile(`^(\b(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(/([1-2]?[0-9]|3[0-2]))?\b[,]?)*$`),
```

Would you accept a PR?

**Comment by ShreyasRmsft** [comment]
We're blocked by this as well

**Comment by Sheludchenko** [comment]
Same thing for me. I added private endpoints to our instances, which immediately blocked public access.

@bart-vmware I'd suggest replacing the current RegEx with 
`
^(\b(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(/([1-2][0-9]|3[0-2]|[3-9]))?\b[,]?)*$
` instead, since the system won't accept any of /2 or /1 anyway.

**Comment by Sheludchenko** [comment]
Added PR for the issue #24306

**Comment by katbyte** [comment]
Thank you for the report, and apologies for the slow follow up here.

This issue is linked to #24306 (ISSUE-18134 - Updated Cosmos DB account validation RegEx for ip_range_filter setting), which merged and shipped in v3.86.0 and appears to have fixed it, so we are closing this one out as resolved.

If you are still running into this on the current (v5.x) version of the provider, please let us know with a comment or open a new issue referencing this one and we will take another look.
